nbi/engine/src/org/netbeans/installer/wizard/containers/SwingFrameContainer.java

```diff
@@ -418,23 +419,28 @@ public void actionPerformed(ActionEvent event) {
     
     private void initializeMacOS() {
         if (SystemUtils.isMacOS()) {
-            final Application application = Application.getApplication();
-            if(application == null) {
-                // e.g. running OpenJDK port via X11 on Mac OS X
-                return;
+            if (!initializeMacJDK("org.netbeans.installer.wizard.containers.initializeMacJDK8")) { // NOI18N
```

Comment:
If there is not much significant difference in look & feel UI , we can then  use Desktop class in JDk8 also instead of Application class.

Reply:
JDK8 Desktop class does not support a QuitHandler.